src/dmd/expressionsem.d

```diff
@@ -7799,6 +7799,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 Expression e = die.semanticY(sc, 1);
                 if (e && isDotOpDispatch(e))
                 {
+                    exp.e2 = exp.e2.expressionSemantic(sc);
```

Comment:
On this branch e2 is semantically analyzed in resolvePropertiesX (below), but that call is done with gagged errors. That is the only time when semantic gets ran on e2, that is why the error never gets to be printed. In order to make sure that UFCS is tried with correct parameters, e2 needs to have semantic ran on it.

Reply:
Thanks! Shouldn't this explanation be in a comment or something though?

Reply:
Done.